src/rvs_blas.cpp

```diff
@@ -1148,6 +1148,32 @@ void rvs_blas::generate_random_matrix_data(void) {
           return;
         }
 
+        if(hipStreamSynchronize(hip_stream) != hipSuccess) {
+          std::cout << "hipStreamSynchronize() failed !!! for stream " << hip_stream << std::endl;
+          is_error = true;
+          return;
+        }
+      }
+      else if((ops_type == "sgemm") || (data_type == "fp32_r")) {
+
+        if(hiprandGenerateUniform(hiprand_generator, (float *)da, size_a) != HIPRAND_STATUS_SUCCESS) {
+          std::cout << "\n hiprandGenerateUniform() failed !!!" << "\n";
+          is_error = true;
+          return;
+        }
+
+        if(hiprandGenerateUniform(hiprand_generator, (float *)db, size_b) != HIPRAND_STATUS_SUCCESS) {
```

Comment:
Maybe add for which matrix the failure in API happened, so that error logs point at right data.